### PR TITLE
Controllers Rspec PR

### PR DESCRIPTION
**Pull request by this-ezzy**
In this milestone, I implemented the following requirements

- Add RSpec as a gem to your project.
- Create a Request spec file for all your controllers.
- Make sure that for each action you check:
> If response status was correct.
> If a correct template was rendered.
> If the response body includes correct placeholder text.
- Ensured no linter errors

### CODE REVIEW

**Comment by prolajumokeoni** [comment]
Hi @Ezzywealth 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted  the wrong pull request for this milestone 👍🏿 
Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/requests/posts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    end
+
+    it 'renders the correct template' do
+      get "/users/#{user.id}/posts"
+      expect(response).to render_template('index')
+    end
+
+    it 'response body includes the correct content' do
+      get "/users/#{user.id}/posts"
+      expect(response.body).to match(%r{<h2>List of posts by a given user</h2>})
```
Comment (mukhammadrizooff) [inline spec/requests/posts_controller_spec.rb]: - [ ] [REQUIRED] Good job so far! Kindly `index` and `show` actions for `Post` controllers should be tested like that:
```ruby
describe 'GET /index' do
  ...
end

describe 'GET /show' do
  ...
end
```

**Review by mukhammadrizooff — CHANGES_REQUESTED** [review]
Hi @Ezzywealth, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter errors ✔️ 
- Good implementations ✔️ 
- Correct GitFlow ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @Ezzywealth, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitfllow. :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**